eZ/Publish/Core/MVC/Symfony/View/Builder/ContentViewBuilder.php

```diff
@@ -0,0 +1,181 @@
+<?php
+
+/**
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Publish\Core\MVC\Symfony\View\Builder;
+
+use eZ\Publish\API\Repository\Repository;
+use eZ\Publish\API\Repository\Values\Content\Content;
+use eZ\Publish\API\Repository\Values\Content\Location;
+use eZ\Publish\API\Repository\Values\Content\VersionInfo;
+use eZ\Publish\Core\Base\Exceptions\InvalidArgumentException;
+use eZ\Publish\Core\Base\Exceptions\UnauthorizedException;
+use eZ\Publish\Core\MVC\Symfony\View\ContentView;
+use eZ\Publish\Core\MVC\Symfony\Security\Authorization\Attribute as AuthorizationAttribute;
+use eZ\Publish\Core\MVC\Symfony\View\ParametersInjector;
+use Symfony\Component\HttpKernel\Controller\ControllerReference;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\Security\Core\Exception\AccessDeniedException;
+use Symfony\Component\Security\Core\Authorization\AuthorizationCheckerInterface;
+
+/**
+ * Builds ContentView objects.
+ */
+class ContentViewBuilder implements ViewBuilder
+{
+    /** @var \eZ\Publish\API\Repository\Repository */
+    private $repository;
+
+    /** @var AuthorizationCheckerInterface */
+    private $authorizationChecker;
+
+    /** @var ParametersInjector */
+    private $viewParametersInjector;
+
+    public function __construct(
+        Repository $repository,
+        AuthorizationCheckerInterface $authorizationChecker,
+        ParametersInjector $viewParametersInjector
+    ) {
+        $this->repository = $repository;
+        $this->authorizationChecker = $authorizationChecker;
+        $this->viewParametersInjector = $viewParametersInjector;
+    }
+
+    public function matches($argument)
+    {
+        return strpos($argument, 'ez_content:') !== false;
+    }
+
+    /**
+     * @throws InvalidArgumentException If both contentId and locationId parameters are missing
+     * @throws NotFoundHttpException If the location is invisible
+     */
+    public function buildView(array $parameters)
+    {
+        $view = new ContentView(null, [], $parameters['viewType']);
+
+        if (isset($parameters['locationId'])) {
+            $location = $this->loadLocation($parameters['locationId']);
+            if ($location->invisible) {
+                throw new NotFoundHttpException('Location cannot be displayed as it is flagged as invisible.');
```

Comment:
Not sure about the Http exception here. We might be in another context (e.g. CLI).